comp/core/gui/guiimpl/systray/Sources/Logger.swift

```diff
@@ -0,0 +1,67 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+import Foundation
+import os.log
+
+enum LogLevel {
+    case debug, info, error
+}
+
+class Logger {
+    private static var useFileLogging = false
+    private static let osLog = OSLog(subsystem: "com.datadoghq.agent", category: "GUI")
+
+    /// Configure logging mode at startup
+    /// - Parameter useFileLogging: If true, uses NSLog (file-based); if false, uses os_log (unified logging)
+    static func configure(useFileLogging: Bool) {
+        self.useFileLogging = useFileLogging
+        if useFileLogging {
+            NSLog("[Logger] File-based logging enabled")
+        } else {
+            os_log("[Logger] Unified logging (os_log) enabled", log: osLog, type: .info)
+        }
+    }
+
+    /// Log a message with specified level and context
+    /// - Parameters:
+    ///   - message: The message to log
+    ///   - level: Log level (debug, info, error)
+    ///   - context: Optional context string (e.g., class name)
+    static func log(_ message: String, level: LogLevel = .info, context: String = "") {
+        let prefix = context.isEmpty ? "" : "[\(context)] "
```

Comment:
I think logLevel parameter alone is not enough, one need a current level somewhere which if logLevel is above than the log will be dropped.